ftp_blueprints/download_file.py

```diff
@@ -128,33 +133,35 @@ def determine_destination_name(
     return destination_name
 
 
-def find_ftp_file_names(client, prefix=''):
+def find_files_in_directory(
+        client,
+        folder_filter,
+        files,
+        folders):
     """
-    Fetched all the files in the folder on the FTP server
+    Pull in a list of all entities under a specific directory and categorize them into files and folders.
     """
-    try:
-        data = []
-        files = []
-        folders = []
-        client.dir(prefix, data.append)
-        for d in data:
-            if d.startswith('d'):
-                folders.append(d.split()[-1])
-            else:
-                name = d.split()[-1]
-                if prefix != '':
-                    files.append(f'{prefix}/{name}')
-                else:
-                    files.append(name)
-        for folder in folders:
-            if prefix:
-                folder = f'{prefix}/{folder}'
-            files.extend(find_ftp_file_names(client, folder))
-    except Exception as e:
-        print(f'Failed to find files in folder {prefix}')
-        raise(e)
+    original_dir = client.pwd()
+    names = client.nlst(folder_filter)
+    for name in names:
+        # Accounts for an issue where some FTP servers return file names
+        # without folder prefixes.
+        if '/' not in name:
+            name = f'{folder_filter}/{name}'
+
+        try:
+            client.cwd(name)
```

Comment:
The moving around with `cwd` feels hacky. Can something like `os.path.isdir` or `ftputil` be used instead?

Reply:
We can probably shift to ftputil in the future, but for right now, using NLST requires us to use this method to determine if something is a file or directory.

Reply:
Okay, then I think it should just be moved into a `finally` block for clarity.